### ISSUE TITLE
Rexray 0.6.3 does not display scaleio snapshots

### ISSUE THREAD

**Issue by rafakei** [codedellemc/rexray#688]
Hi, RR Dev, 

Rexray 0.6.3 does not display scaleio snapshots, please help.

          #	rexray -h

	Available Commands:
	  version     Print the version
	  env         Print the REX-Ray environment
	  adapter     The adapter manager
	  device      The device manager
	  volume      The volume manager
	  service     The service controller
	  help        Help about any command

	# rexray volume ls
	ID                Name        Status       Size
	ae103aa900000000  prod1-gold  unavailable  8

	# scli --query_all_volumes
	Query-all-volumes returned 2 volumes
	Protection Domain 72d50fd900000000 Name: PD1
	Storage Pool 519fd36700000000 Name: SP1
	 Volume ID: ae103aa900000000 Name: prod1-gold Size: 8.0 GB (8192 MB) Mapped to 1 SDC Thin-provisioned
	 Volume ID: ae103aaa00000001 Name: prodvol1 Size: 8.0 GB (8192 MB) Unmapped Snapshot of prod1-gold

	# rexray version
	REX-Ray
	-------
	Binary: /usr/bin/rexray
	SemVer: 0.6.3
	OsArch: Linux-x86_64
	Branch: v0.6.3
	Commit: 69b1f5c2d86a2103c792bec23b5855babada1c0a
	Formed: Thu, 08 Dec 2016 07:22:14 CST

	libStorage
	----------
	SemVer: 0.3.5
	OsArch: Linux-x86_64
	Branch: v0.6.3
	Commit: 456dd68123dd6b49da0275d9bbabd6c800583f61
	Formed: Thu, 08 Dec 2016 07:21:36 CST

	# docker version
	Client:
	 Version:      1.12.5
	 API version:  1.24
	 Go version:   go1.6.4
	 Git commit:   7392c3b
	 Built:        Fri Dec 16 02:23:59 2016
	 OS/Arch:      linux/amd64

	Server:
	 Version:      1.12.5
	 API version:  1.24
	 Go version:   go1.6.4
	 Git commit:   7392c3b
	 Built:        Fri Dec 16 02:23:59 2016
	 OS/Arch:      linux/amd64

	# cat /etc/rexray/config.yml
	libstorage:
	  service: scaleio
	  integration:
		volume:
		  operations:
			create:
			  default:
				size: 8
		  mount:
			  preempt: true
	scaleio:
	  insecure: true
	  thinOrThick: ThinProvisioned
	  endpoint: https://10.32.197.170/api
	  userName: admin
	  password: ABcd1234
	  systemName: sio5
	  protectionDomainName: PD1
	  storagePoolName: SP1
	  
	#  rexray volume ls -l debug

	DEBU[0000] built output template                         host=unix:///var/run/libstorage/132486467.sock template={{define "printObject"}}{{printf "%v" .}}{{end}}{{define "printPrettyJSON"}}{{. | jsonp}}{{end}}{{define "printEncVolumeWithPathFields"}}{{.ID}}      {{.Name}}       {{.Volume.AttachmentState | printAttState}}     {{.Size}}       {{.Encrypted}}  {{.Path}}{{end}}{{define "printSnapshotFields"}}{{.ID}}{{.Name}}       {{.Status}}     {{.VolumeID}}{{end}}{{define "printInstanceFields"}}{{.InstanceID.ID}}  {{.Name}}       {{.ProviderName}}       {{.Region}}{{end}}{{define "printServiceFields"}}{{.Name}}    {{.Driver.Name}}{{end}}{{define "printStringSlice"}}{{.}}{{end}}{{define "printJSON"}}{{. | json}}{{end}}{{define "printVolumeID"}}{{.ID}}{{end}}{{define "printVolumeFields"}}{{.ID}}   {{.Name}}       {{.AttachmentState | printAttState}}   {{.Size}}{{end}}{{define "printVolumeWithPathFields"}}{{.ID}}   {{.Name}}       {{.Volume.AttachmentState | printAttState}}     {{.Size}}       {{.Path}}{{end}}{{define "printEncVolumeFields"}}{{.ID}}       {{.Name}}       {{.AttachmentState | printAttState}}    {{.Size}}       {{.Encrypted}}{{end}}{{define "printMountFields"}}{{.ID}}       {{.Source}}     {{.MountPoint}}{{end}}ID       Name    Status  Size
	{{range sort .D "Name" }}{{template "printVolumeFields" .}}
	{{end}} time=1483605723060
	ID                Name        Status       Size
	ae103aa900000000  prod1-gold  unavailable  8
	DEBU[0000] waiting until libStorage is stopped           host=unix:///var/run/libstorage/132486467.sock time=1483605723061
	DEBU[0000] done waiting on err chan; err chan is nil     host=unix:///var/run/libstorage/132486467.sock time=1483605723061
	DEBU[0000] completed cli execution                       time=1483605723061
	INFO[0000] exiting process                               time=1483605723061
	DEBU[0000] completed onExit at end of program            time=1483605723061

**Comment by kacole2** [comment]
i think @cduchesne can be the source of truth on this, but snapshots haven't been implemented in the latest releases. It was only working on v0.3.3. Once he confirms, it would be good to open a feature request on [libStorage](https://github.com/codedellemc/libstorage/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aopen)

**Comment by kacole2** [comment]
this looks to be a duplicate of #345 #616

**Comment by rafakei** [comment]
@cduchesne 
Please help with this issue. It seems more like a code issue, because when base volume is removed, the shallowest snapshots all appear under "rexray volume ls":

	# scli --mdm_ip 10.0.1.179 --query_all_volumes
	Query-all-volumes returned 4 volumes
	Protection Domain 72d50fd900000000 Name: PD1
	Storage Pool 519fd36700000000 Name: SP1
	 Volume ID: ae103ac500000000 Name: vol1 Size: 104.0 GB (106496 MB) Unmapped Thick-provisioned
	 Volume ID: ae103ac600000001 Name: vol1-snap-1 Size: 104.0 GB (106496 MB) Unmapped Snapshot of vol1
	 Volume ID: ae103ac800000002 Name: vol1-snap-2 Size: 104.0 GB (106496 MB) Unmapped Snapshot of vol1
	 Volume ID: ae103ac900000003 Name: vol1-snap-2-snap-1 Size: 104.0 GB (106496 MB) Unmapped Snapshot of vol1-snap-2

	# rexray volume ls
	ID                Name  Status     Size
	ae103ac500000000  vol1  available  104


	# scli --mdm_ip 10.0.1.179 --remove_volume --volume_name vol1 --i_am_sure
	Successfully removed volume 'vol1'

	# scli --mdm_ip 10.0.1.179 --query_all_volumes
	Query-all-volumes returned 3 volumes
	Protection Domain 72d50fd900000000 Name: PD1
	Storage Pool 519fd36700000000 Name: SP1
	 Volume ID: ae103ac600000001 Name: vol1-snap-1 Size: 104.0 GB (106496 MB) Unmapped Snapshot of a deleted volume
	 Volume ID: ae103ac800000002 Name: vol1-snap-2 Size: 104.0 GB (106496 MB) Unmapped Snapshot of a deleted volume
	 Volume ID: ae103ac900000003 Name: vol1-snap-2-snap-1 Size: 104.0 GB (106496 MB) Unmapped Snapshot of vol1-snap-2
	 
	# rexray volume ls
	ID                Name         Status     Size
	ae103ac600000001  vol1-snap-1  available  104
	ae103ac800000002  vol1-snap-2  available  104

**Comment by cduchesne** [comment]
You're seeing snapshots in the volume list because that's what the ScaleIO API returns via the volume list when the primary volume is removed and there are snapshots.

See below:
```
[root@rancher-phoenix ~]# curl -ks -u :<key> https://<api-server>/api/types/Volume/instances | jq -M '.' | grep name
    "name": "vol02",
    "name": "vol03",
    "name": "vol01",
    "name": "vol04-snap-1",
```

**Comment by cduchesne** [comment]
In my testing, the volume snapshot is also usable via rexray and docker.

We have not yet implemented snapshots with the current version of REX-Ray. We made significant changes moving from REX-Ray 0.3 to 0.4 and snapshot feature/functionality has not been prioritized as of yet. Is there a specific use-case you're looking to solve that specifically requires snapshots?

**Comment by rafakei** [comment]
@cduchesne 
Yes, it is important. Customer hopes Rexray can manipulate a snapshot just like a volume, so that they can quickly "clone" volumes and have them attach to different testing containers.

**Comment by rafakei** [comment]
@cduchesne The most needed feature is to "rexray ls" and "rexray attach" a snapshot just like a volume. Customer can use remote scli to achieve "rexray snapshot create/delete"

**Comment by koensayr** [comment]
Consider moved and thus is being tracked below. Going to close it here.

https://github.com/codedellemc/rexray/issues/616
